packages/mdc-chips/chip-set/foundation.js

```diff
@@ -55,42 +56,60 @@ class MDCChipSetFoundation extends MDCFoundation {
     super(Object.assign(MDCChipSetFoundation.defaultAdapter, adapter));
 
     /**
-     * The selected chips in the set. Only used for choice chip set or filter chip set.
-     * @private {!Array<!MDCChipFoundation>}
+     * The ids of the selected chips in the set. Only used for choice chip set or filter chip set.
+     * @private {!Array<string>}
      */
-    this.selectedChips_ = [];
+    this.selectedChipIds_ = [];
   }
 
   /**
-   * Selects the given chip. Deselects all other chips if the chip set is of the choice variant.
-   * @param {!MDCChipFoundation} chipFoundation
+   * Returns an array of the IDs of all selected chips.
+   * @return {!Array<string>}
    */
-  select(chipFoundation) {
-    if (this.adapter_.hasClass(cssClasses.CHOICE)) {
-      this.deselectAll_();
+  getSelectedChipIds() {
+    return this.selectedChipIds_;
+  }
+
+  /**
+   * Toggles selection of the chip with the given id.
+   * @param {string} chipId
+   */
+  toggleSelect(chipId) {
+    if (this.adapter_.hasClass(cssClasses.CHOICE) || this.adapter_.hasClass(cssClasses.FILTER)) {
+      if (this.selectedChipIds_.indexOf(chipId) >= 0) {
+        this.deselect(chipId);
+      } else {
+        this.select(chipId);
+      }
     }
-    chipFoundation.setSelected(true);
-    this.selectedChips_.push(chipFoundation);
   }
 
   /**
-   * Deselects the given chip.
-   * @param {!MDCChipFoundation} chipFoundation
+   * Selects the chip with the given id. Deselects all other chips if the chip set is of the choice variant.
+   * @param {string} chipId
    */
-  deselect(chipFoundation) {
-    const index = this.selectedChips_.indexOf(chipFoundation);
-    if (index >= 0) {
-      this.selectedChips_.splice(index, 1);
+  select(chipId) {
+    if (this.adapter_.hasClass(cssClasses.CHOICE)) {
+      const selectedChipId = this.selectedChipIds_.length > 0 && this.selectedChipIds_[0];
```

Comment:
Closure is not a fan of the typing here as it thinks selectedChipId could evaluate to boolean or string (although I don't think it would ever actually make it into the setSelected method as a boolean). Maybe just only go inside this if when this.adapter_.hasClass(cssClasses.CHOICE) && this.selectedChipIds_.length > 0 ?